app/src/main/java/org/groundplatform/android/model/geometry/Geometry.kt

```diff
@@ -92,13 +93,27 @@ data class LineString(val coordinates: List<Coordinates>) : Geometry {
   override val area: Double
     get() = 0.0
 
+  val tooltipDistance: Double?
+    get() {
+      if (coordinates.size < 2 || isClosed()) return null
+      val lastVertex = coordinates.last()
+      val secondLastVertex = coordinates[coordinates.size - 2]
+      return calculateDistanceInMeters(secondLastVertex, lastVertex)
+    }
+
   override fun center(): Coordinates = coordinates.centerOrError()
 
   override fun isEmpty() = coordinates.isEmpty()
 
   fun isClosed(): Boolean =
     coordinates.size >= 4 && coordinates.firstOrNull() == coordinates.lastOrNull()
 
+  private fun calculateDistanceInMeters(start: Coordinates, end: Coordinates): Double {
```

Comment:
This isn't specific to Geometry; perhaps move into an existing Ext.kt file as `Coordinates.getDistanceInMeters`?